src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -911,22 +911,22 @@ private static Frame findCallingMethodFrame() {
 
         @Specialization
         public DynamicObject threadWaitFDNode(int fd) {
-            final FDSet fdSet = getContext().getNativePlatform().createFDSet();
-
-            getContext().getThreadManager().runBlockingSystemCallUntilResult(this, () -> {
-                fdSet.set(fd);
-                final int result = nativeSockets().select(fd + 1,
-                        fdSet.getPointer(),
-                        Pointer.JNR_NULL,
-                        Pointer.JNR_NULL,
-                        null);
-
-                if (result == 0) {
-                    return null;
-                }
+            try (FDSet fdSet = getContext().getNativePlatform().createFDSet()) {
+                getContext().getThreadManager().runBlockingSystemCallUntilResult(this, () -> {
+                    fdSet.set(fd);
+                    final int result = nativeSockets().select(fd + 1,
+                            fdSet.getPointer(),
+                            Pointer.JNR_NULL,
+                            Pointer.JNR_NULL,
+                            null);
+
+                    if (result == 0) {
```

Comment:
I'm curious when this happens without a timeout? Is it an interrupt? If so, it should retry on interrupt?

Reply:
Yeah I think this basically cannot happen without a timeout.
A signal interrupt will return -1 and set EINTR.